### ISSUE TITLE
resize_image_with_crop_or_pad doesn't work with input pipelines

### ISSUE THREAD

**Issue by ry** [tensorflow/tensorflow#521]
Because `tf.image.resize_image_with_crop_or_pad()` requires its input to have a fully defined shape, it's not useful as part of an input pipeline since the size of images loaded will not be known in advance.

``` python
  queue = tf.train.string_input_producer(filenames)
  reader = tf.WholeFileReader()
  _, contents = reader.read(queue)
  image = tf.image.decode_jpeg(contents, channels=3)
  cropped = tf.image.resize_image_with_crop_or_pad(image, 224, 224)
```

Ideally the above would work...

**Comment by martinwicke** [comment]
Yes, that's a bug.

**Comment by eerwitt** [comment]
I needed to use the method `resize_image_with_crop_or_pad` as part of an input pipeline and ran into a similar issue.

I'm not yet proficient enough with TensorFlow to tell if this is useful. I made a small workaround for personal use. If this workaround would be a route to use in fixing this issue I'd be happy to update the tests, docs and change out the logic in the existing implementation. If anyone can advise on its usefulness I'd appreciate the feedback.

[Example Implementation of resize_image_with_crop_or_pad](https://gist.github.com/eerwitt/51aba4bffd9ddd5c581c#file-resize_image_with_crop_or_pad_pipeline-py-L6)

**Comment by po0ya** [comment]
@eerwitt It works flawlessly! thanks.

**Comment by martinwicke** [comment]
Send a PR?

**Comment by kYroL01** [comment]
So any news to solve the problem ?

**Comment by martinwicke** [comment]
Other than the workaround posted above, no.

**Comment by gaohuazuo** [comment]
Here is a monkey patching module based on PR https://github.com/tensorflow/tensorflow/pull/2737

[resize_image_patch.py](https://gist.github.com/gaohuazuo/4093eee2aee4d5b8746bff698506ee57#file-resize_image_patch-py)

**Note**
* Remember to pass argument `dynamic_shape=True`
* Input image must have static rank of 3. This can be done by `tf.Tensor.set_shape([None, None, None])`

**Comment by yongtang** [comment]
Looks like the issue has been fixed by PR #2737. Maybe this issue could be closed?

**Comment by martinwicke** [comment]
Thanks!
